src/main/java/org/openrewrite/java/migrate/lang/var/UseVarForObject.java

```diff
@@ -0,0 +1,171 @@
+        List<RecipeExample.Source> examples = new ArrayList<>();
+        //language=java
+        examples.add(new RecipeExample.Source(
+                "package com.example.app;\n" +
+                "\n" +
+                "class A {\n" +
+                "  void m() {\n" +
+                "    Object o = new Object();\n" +
+                "  }\n" +
+                "}",
+                "package com.example.app;\n" +
+                "\n" +
+                "class A {\n" +
+                "  void m() {\n" +
+                "    var o = new Object();\n" +
+                "  }\n" +
+                "}",
+                null, "java"));
+
+        //language=java
+        examples.add(new RecipeExample.Source(
+                "package com.example.app;\n" +
+                "\n" +
+                "class A {\n" +
+                "  Object o = new Object();\n" +
+                "  void m() {\n" +
+                "    Object inner = o;\n" +
+                "  }\n" +
+                "}",
+                "package com.example.app;\n" +
+                "\n" +
+                "class A {\n" +
+                "  void m() {\n" +
+                "    var inner = o;\n" +
+                "  }\n" +
+                "}",
+                null, "java"));
+        //language=java
+        examples.add(new RecipeExample.Source(
+                "package com.example.app;\n" +
+                "\n" +
+                "class A {\n" +
+                "  static {\n" +
+                "    Object o = new Object();\n" +
+                "  }\n" +
+                "}",
+                "package com.example.app;\n" +
+                "\n" +
+                "class A {\n" +
+                "  static {\n" +
+                "    var o = new Object();\n" +
+                "  }\n" +
+                "}",
+                null, "java"));
+        //language=java
+        examples.add(new RecipeExample.Source(
+                "package com.example.app;\n" +
+                "\n" +
+                "class A {\n" +
+                "  {\n" +
+                "    Object o = new Object();\n" +
+                "  }\n" +
+                "}",
+                "package com.example.app;\n" +
+                "\n" +
+                "class A {\n" +
+                "  {\n" +
+                "    var o = new Object();\n" +
+                "  }\n" +
+                "}",
+                null, "java"));
+
+        //language=markdown
+        recipeExample.setDescription("Applies `var` keyword to primitive variable definitions if the type is different from `short` or `byte`");
+        recipeExample.setSources(examples);
+
+        List<RecipeExample> exampleList = new ArrayList<>();
+        exampleList.add(recipeExample);
+        return exampleList;
+    }
+
+    @NotNull
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new HasJavaVersion("10", true),
```

Comment:
We typically use `UsesJavaVersion` instead. Not sure why we have both actually...